Core/clim-core/panes/stream-panes.lisp

```diff
@@ -365,49 +365,50 @@ current background message was set."))
-      (loop  for (key value) on options by #'cddr
-             if (and (member key space-keys :test #'eq)
-                     (not (eq value :compute)))
-               nconc (list key value) into space-options
-             else
-               nconc (list key value) into other-options
-             end
-             finally (progn
-                       (setq user-sr space-options)
-                       (setq pane-options other-options)))
-      (let* ((pane (apply #'make-pane type (append pane-options
-                                                   (unless (or scroll-bars
-                                                               borders)
-                                                     user-sr))))
-             (stream pane))
-        (when scroll-bars
-          (setq pane (apply #'make-pane 'scroller-pane
-                            :scroll-bar scroll-bars
-                            :contents (list (make-pane 'viewport-pane
-                                                       :contents (list pane)))
-                            (unless borders
-                              user-sr))))
-        (when borders
-          (setq pane (apply #'make-pane 'border-pane
-                            :border-width (if (not (numberp borders))
-                                              1
-                                              borders)
-                            :contents (list pane)
-                            user-sr)))
-        (values pane stream)))))
+  (with-keywords-removed (options (:type :scroll-bar :scroll-bars :borders :label
+                                         :label-alignment :display-after-commands))
+    ;; See Spec 29.4.3 for :display-after-commands keyword.
+    (when (and display-after-commands-p (not (member :display-time options)))
+      (if (eq t display-after-commands)
+          (nconc options (list :display-time :command-loop))
+          (nconc options (list :display-time display-after-commands))))
+    (let* ((pane (apply #'make-pane type options))
+           (stream pane))
+      (when scroll-bars
+        (setq pane (apply #'make-pane 'scroller-pane
+                          :contents (list (make-pane 'viewport-pane
+                                                     :contents (list pane)))
+                          ;; From the Franz manual if :scroll-bars is a
+                          ;; cons the car is treated as the non-cons
+                          ;; argument and the cdr is a list of keyword
+                          ;; argument pairs to be used as options of
+                          ;; the scroller-pane
+                          (if (consp scroll-bars)
+                              `(:scroll-bar ,@scroll-bars)
+                              `(:scroll-bar ,scroll-bars)))))
+      (when label
+        (setq pane (apply #'make-pane 'label-pane
+                          :label label
+                          :label-alignment label-alignment
```

Comment:
This could pass `:label-alignment nil` which seems forbidden:

> … label-alignment may be either :bottom or :top, …

Reply:
I have added label-alignment-p  to test if the keyword is supplied.